### PR TITLE
Useful methods & traits

### PR DESCRIPTION
**Pull request by ThomasdenH**
Implement some useful methods and traits. For types where there is only one sensible implementation, I have implemented `Display`. There is also some documentation with examples.

Notable:
- `Amount` implements a partial ordering when the currency is the same.
- I have moved the String -> Date conversions to use explicit 'unchecked' in their name.

### CODE REVIEW

File: beancount-core/src/account_types.rs [hunk 1]
```diff
@@ -1,11 +1,35 @@
...
+    /// # Example
+    /// ```rust
+    /// use beancount_core::AccountType::*;
+    /// assert_eq!(Assets.default_name(), "Assets");
+    /// assert_eq!(Liabilities.default_name(), "Liabilities");
+    /// assert_eq!(Equity.default_name(), "Equity");
+    /// assert_eq!(Income.default_name(), "Income");
+    /// assert_eq!(Expenses.default_name(), "Expenses");
+    /// ```
+    pub fn default_name(&self) -> &'static str {
```
Comment (twilco) [inline beancount-core/src/account_types.rs:25]: Small nit: the word "default" implies there are other "types" of names.  I'm guessing that is false?  If so, would `name` (or something else) work here?

File: beancount-core/src/amount.rs [hunk 2]
```diff
@@ -15,8 +15,18 @@ pub struct Amount<'a> {
...
 
+impl cmp::PartialOrd for Amount<'_> {
+    fn partial_cmp(&self, other: &Amount<'_>) -> Option<cmp::Ordering> {
+        if self.currency == other.currency {
+            self.num.partial_cmp(&other.num)
+        } else {
+            None
+        }
+    }
+}
```
Comment (twilco) [inline beancount-core/src/amount.rs:26]: This seems really useful!

File: beancount-core/src/account_types.rs (hunk 1 above)
Comment (twilco) [reply beancount-core/src/account_types.rs:25]: Nevermind, I just got to the part of the PR mentioning account name changes via `option`.

**Review by twilco — APPROVED** [review]
This is a great PR.  Thanks so much!

**Comment by twilco** [comment]
I merged this, but @mbudde feel free to leave any comments and we can address in a follow-up PR.

File: beancount-core/src/account_types.rs (hunk 1 above)
Comment (ThomasdenH) [reply beancount-core/src/account_types.rs:25]: Still, it turns out that it's useful to have that in the documentation here.

**Pull request merged**